src/prefect/workers/base.py

```diff
@@ -487,6 +487,7 @@ async def teardown(self, *exc_info):
 
     async def get_and_submit_flow_runs(self):
         runs_response = await self._get_scheduled_flow_runs()
+        self._emit_worker_heartbeat_event(heartbeat_origin="get_and_submit_flow_runs")
```

Comment:
we could wait til this submits to emit the event, but i figure if we're pollin, we're rollin

Reply:
Makes sense to me.